src/endpoint/iam/ops/iam_list_users.js

```diff
@@ -12,10 +13,11 @@ async function list_users(req, res) {
 
     const params = {
         marker: req.body.marker,
-        max_items: req.body.max_items,
+        max_items: req.body.max_items ?? iam_constants.DEFAULT_MAX_ITEMS,
```

Comment:
i am not sure that nullish condition is better than simple logical or. what should be done if body max items is 0?

Reply:
I will throw an error because in the docs of [ListUsers](https://docs.aws.amazon.com/IAM/latest/APIReference/API_ListUsers.html#API_ListUsers_RequestParameters): 

> Valid Range: Minimum value of 1. Maximum value of 1000.

I tested it with 0 and saw an empty array (also tested it with a negative value -1 and got the items).
In the code, I decide to act as it is supposed to be according to the docs, and I will throw an error on values out of the range (including 0).
Using the nullish coalescing operator operator gives exactly what I want - according to the docs:

> If you do not include this parameter, the number of items defaults to 100.

Since you raised it now I noticed that for this parameter they used "Valid Range" While in places I tested and saw errors they used "Length Constraints". I still think it is better to throw an error and enforce the values to use as in the expected range.

Reminder - currently we haven't implemented pagination yet, so the parameter is not supported yet

Reply:
ok, you definitely understand the difference of || and ?? well but this is always subtle for many people to read unless carefully focusing on it, imho. in any case we can use it, but I am trying to minimize it.

Reply:
I can add a comment in the code: "If one did not include max_items parameter (req.body.max_items value is undefined), we use a default number DEFAULT_MAX_ITEMS", but I don't like it (it explains the operator ?? and does not give an added value), let me know what you think.

Reply:
oh but what is the type of req.body.max_items? string? if so then we actually want to parse it to a number and not send once a string and once a number (when undefined)

Reply:
It should be a number, but now I noticed that there is an issue with AWS CLI in which it doesn't pass it.

I ran: 
`s3-nc-user-1-iam iam list-users --path-prefix /abc/ --max-items 7 --debug` 
I added the `--path-prefix` flag so we can see a flag that passes (and a flag that I think doesn't pass).
(s3-nc-user-1-iam is an alias 
for `AWS_ACCESS_KEY_ID=<access-key> AWS_SECRET_ACCESS_KEY=<secret-key> aws --no-verify-ssl --endpoint-url https://localhost:7005`).

In the debug I see AWS CLI list the argument of max-items:
```bash
2024-07-07 08:43:35,751 - MainThread - awscli.clidriver - DEBUG - Arguments entered to CLI: ['--no-verify-ssl', '--endpoint-url', 'https://localhost:7005', 'iam', 'list-users', '--path-prefix', '/abc/', '--max-items', '7', '--debug']
```

In the debug I don't see the max params passing to the request's body (PathPrefix we can see):
```bash
2024-07-07 08:43:35,836 - MainThread - botocore.endpoint - DEBUG - Making request for OperationModel(name=ListUsers) with params: {'url_path': '/', 'query_string': '', 'method': 'POST', 'headers': {'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8', 'User-Agent': 'aws-cli/2.15.36 Python/3.11.9 Darwin/23.4.0 source/arm64 prompt/off command/iam.list-users'}, 'body': {'Action': 'ListUsers', 'Version': '2010-05-08', 'PathPrefix': '/abc/'}, 'url': 'https://localhost:7005/', 'context': {'client_region': 'us-east-1', 'client_config': <botocore.config.Config object at 0x1083b49d0>, 'has_streaming_input': False, 'auth_type': None}}
```

Same issue in list-access-keys:
`s3-nc-user-1-iam iam list-access-keys --max-items 1 --user-name shira-1001 --debug`

In the debug I see AWS CLI list the argument of max-items:
```bash
2024-07-07 09:03:15,120 - MainThread - awscli.clidriver - DEBUG - Arguments entered to CLI: ['--no-verify-ssl', '--endpoint-url', 'https://localhost:7005', 'iam', 'list-access-keys', '--max-items', '1', '--user-name', 'shira-1001', '--debug']
```

In the debug I don't see the max params passing to the request's body (UserName we can see):
```bash
2024-07-07 09:03:15,228 - MainThread - botocore.endpoint - DEBUG - Making request for OperationModel(name=ListAccessKeys) with params: {'url_path': '/', 'query_string': '', 'method': 'POST', 'headers': {'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8', 'User-Agent': 'aws-cli/2.15.36 Python/3.11.9 Darwin/23.4.0 source/arm64 prompt/off command/iam.list-access-keys'}, 'body': {'Action': 'ListAccessKeys', 'Version': '2010-05-08', 'UserName': 'shira-1001'}, 'url': 'https://localhost:7005/', 'context': {'client_region': 'us-east-1', 'client_config': <botocore.config.Config object at 0x106fa89d0>, 'has_streaming_input': False, 'auth_type': None}}
```

Meanwhile, I will mention the issue in the code.

Reply:
just making sure - do you have the latest aws cli? perhaps submit an issue to its github project?

Reply:
I updated my `awscli` (using brew) and still face this issue.
I opened a Github issue in their repo:
https://github.com/aws/aws-cli/issues/8800